pkg/template/processor_test.go

```diff
@@ -0,0 +1,604 @@
+		namespace, commit, user := templateVars()
+		values := paramsKeyValues(namespace, commit, user)
+
+		// adding random param
+		random := getNameWithTimestamp("random")
+		values[random] = random
+
+		cl := test.NewFakeClient(t)
+		p := template.NewProcessor(cl, s)
+
+		// when
+		objs, err := p.Process(templateContent(namespaceObj), values)
+
+		// then
+		require.NoError(t, err)
+		require.Len(t, objs, 1)
+		// namespace
+		verifyResource(t, objs[0].Object.GetObjectKind(), namespace, commit, user)
+	})
+
+	t.Run("should process template with default parameters", func(t *testing.T) {
+		// given
+		s := addToScheme(t)
+		// default values
+		commit, user := "123abc", "toolchain-dev"
+
+		namespace := uuid.NewV4().String()
+		values := make(map[string]string)
+		values["PROJECT_NAME"] = namespace
+
+		cl := test.NewFakeClient(t)
+		p := template.NewProcessor(cl, s)
+
+		// when
+		objs, err := p.Process(templateContent(namespaceObj, roleBindingObj), values)
+
+		// then
+		require.NoError(t, err)
+		require.Len(t, objs, 2)
+
+		// namespace
+		verifyResource(t, objs[0].Object.GetObjectKind(), namespace, commit, user)
+
+		// role binding
+		verifyResource(t, objs[1].Object.GetObjectKind(), namespace, user)
+	})
+
+	t.Run("should process template with missing required params", func(t *testing.T) {
+		// given
+		s := addToScheme(t)
+		values := make(map[string]string)
+
+		cl := test.NewFakeClient(t)
+		p := template.NewProcessor(cl, s)
+
+		// when
+		objs, err := p.Process(templateContent(namespaceObj, roleBindingObj), values)
+
+		// then
+		require.Error(t, err, "fail to process as not providing required param PROJECT_NAME")
+		assert.Nil(t, objs)
+	})
+
+	t.Run("should fail to process template for invalid template content", func(t *testing.T) {
+		// given
+		s := addToScheme(t)
+		namespace, commit, user := templateVars()
+		values := paramsKeyValues(namespace, commit, user)
+
+		cl := test.NewFakeClient(t)
+		p := template.NewProcessor(cl, s)
+
+		// when
+		_, err := p.Process([]byte(namespaceObj), values)
+
+		// then
+		assert.Error(t, err)
+	})
+}
+
+func TestFilter(t *testing.T) {
+
+	ns1 := &unstructured.Unstructured{
+		Object: map[string]interface{}{
+			"kind": "Namespace",
+			"metadata": map[string]interface{}{
+				"name": "ns1",
+			},
+		},
+	}
+	ns2 := &unstructured.Unstructured{
+		Object: map[string]interface{}{
+			"kind": "Namespace",
+			"metadata": map[string]interface{}{
+				"name": "ns2",
+			},
+		},
+	}
+	rb1 := &unstructured.Unstructured{
+		Object: map[string]interface{}{
+			"kind": "RoleBinding",
+			"metadata": map[string]interface{}{
+				"name": "rb1",
+			},
+		},
+	}
+	rb2 := &unstructured.Unstructured{
+		Object: map[string]interface{}{
+			"kind": "RoleBinding",
+			"metadata": map[string]interface{}{
+				"name": "rb2",
+			},
+		},
+	}
+
+	t.Run("filter namespaces", func(t *testing.T) {
+
+		t.Run("return one", func(t *testing.T) {
+			// given
+			objs := []runtime.RawExtension{
```

Comment:
for me, the short version:
```go
objs := []runtime.RawExtension{{Object: ns1}, {Object: rb1}}
```
if more readable 
WDYT?

Reply:
I prefer using separate lines, but I guess it's a matter of taste

Reply:
Then keep it as it is ;-),

Reply:
ok ;)